### DISCUSSION TITLE
Content Security Policy directive: "script-src 'self' Error

### DISCUSSION THREAD

**Discussion by rreeves8** [remix-run/remix#6706] [Q&A] [answered]
Hi, 

What is the standard way to set content policy restrictions for remix applications ? I saw there was a comment on an earlier thread however it seemed to not be as secure as I would hope for. More specially in `(helmet)[https://helmetjs.github.io/]` running on express apps. 

I'm currently getting this error -> Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'self'

**Comment by xHomu** [comment, marked as answer]
Take a look at how the Epic Stack setup CSP:

https://github.com/epicweb-dev/epic-stack/blob/main/server/index.ts#L92
https://github.com/epicweb-dev/epic-stack/blob/main/app/root.tsx#L185
